File: leetcode/1313.py
"""
Runtime: 64 ms, faster than 84.90% of Python3 online submissions for Decompress Run-Length Encoded List.
Memory Usage: 14.9 MB, less than 26.59% of Python3 online submissions for Decompress Run-Length Encoded List.
"""


class Solution:
    def decompressRLElist(self, nums: List[int]) -> List[int]:
        res = []
        while nums:
            freq, val = nums.pop(0), nums.pop(0)
            res += freq * [val]

        return res

File: leetcode/1313.cpp
// Runtime: 3 ms, faster than 94.46% of C++ online submissions for Decompress Run-Length Encoded List.
// Memory Usage: 10.1 MB, less than 44.81% of C++ online submissions for Decompress Run-Length Encoded List.
class Solution {
public:
  vector<int> decompressRLElist(vector<int> &nums) {
    vector<int> res = {};
    for (int i = 0; i < nums.size() / 2; ++i) {
      int freq = nums[2 * i], value = nums[2 * i + 1];
      for (int j = 0; j < freq; ++j)
        res.push_back(value);
    }

    return res;
  }
};
